File: agent/bindings/csharp/Position.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class Position
  {
    [JsonProperty(PropertyName = "line")]
    public int Line { get; set; }
    [JsonProperty(PropertyName = "character")]
    public int Character { get; set; }
  }
}

File: vscode/src/jsonrpc/agent-protocol-factory/position.ts
import assert from 'node:assert'
import * as vscode from 'vscode'
import type * as protocol from '../agent-protocol'

export namespace Position {
    export function from(position: vscode.Position): protocol.Position {
        return {
            line: position.line,
            character: position.character,
        }
    }

    export function vsc(position: protocol.Position): vscode.Position {
        assert(position.line >= 0 && position.character >= 0)
        return new vscode.Position(position.line, position.character)
    }
}
